### ISSUE TITLE
Language syntax ambiguities for "method:" selector notation

### ISSUE THREAD

**Issue by brianlheim** [supercollider/supercollider#2568]
<img width="110" alt="screen shot 2016-12-18 at 6 36 41 pm" src="https://cloud.githubusercontent.com/assets/15369640/21297606/f32df630-c550-11e6-8a12-c78075123fd0.png">

The first line evaluates to 16, and the second line throws an error because the `16s` is immediately understood as 16.1, leaving the garbage method selector `qrsum:`. This could be a problem with the language syntax, or the IDE highlighting, or both.

**Comment by snappizz** [comment]
This appears to be unique to the scale degree syntax. It doesn't happen for the `pi` suffix:

    36 pins: 4 // Message 'pins' not understood
    36pins:4 // Message 'pins' not understood

Radix notation is weird but its result is not too surprising:

    36 rungs: 3 // Message 'rungs' not understood
    36rungs:3 // Unexpected ':'

I don't think it's worth the breaking change to alter the syntax, since 1) putting a method selector after a number literal with no whitespace is only useful in code golfing and 2) almost nobody even knows about the scale degree syntax. So I'd argue that it's a highlighting issue to be lumped in with #2545.

**Comment by jamshark70** [comment]
IMO we should reject all use of infix method names that are not preceded by whitespace. The compiler bug is that it doesn't consistently fail this case.

**Comment by brianlheim** [comment]
I agree with @jamshark70. This seems to me like a fairly significant issue with compiler tokenization.
